compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/dfa/FirDataFlowAnalyzer.kt

```diff
@@ -463,22 +447,22 @@ abstract class FirDataFlowAnalyzer<FLOW : Flow>(
                 if (operandVariable.isReal()) {
                     flow.addTypeStatement(operandVariable typeEq type)
                 }
-                logicSystem.approveStatementsInsideFlow(
-                    flow,
-                    operandVariable notEq null,
-                    shouldRemoveSynthetics = true,
-                    shouldForkFlow = false
-                )
+                if (!type.canBeNull) {
+                    flow.assumeNotNull(operandVariable, shouldForkFlow = false, shouldRemoveSynthetics = true)
```

Comment:
That's extremely strange that this change fixed the test, because there is no single `as` in the test. Could you please investigate real reason of test change? Maybe it's because of previous commits?

Reply:
Is the test you're talking about the one with `o ?: true`? I've accidentally grouped two changes into one commit again, sorry. It's fixed by the part that auto-translates all `notEq null` statements into `typeEq any`. Split into two commits:

https://github.com/JetBrains/kotlin/pull/5017/commits/b347e5b92a28d02fd3a3e7934c6177d3939c324b
https://github.com/JetBrains/kotlin/pull/5017/commits/11d1713f6d66d8177244946addba3811247ad360